### ISSUE TITLE
CupertinoSliverNavigationBar should overscroll inside the bar, not between it and the next sliver

### ISSUE THREAD

**Issue by jslavitz** [flutter/flutter#25061]
Currently, when overscrolling a cupertino sliver nav bar, the contents are pulled away from the nav bar, whereas on iOS, the nav bar is supposed to be pulled down with the contents. You can see the difference by overscrolling in the navigation demo of flutter gallery and the native settings app.

**Comment by xster** [comment]
From duped #28621, we need to make sure the large title font size grows as we overscroll too.

**Comment by stellanhaglund** [comment]
Any updates on this?

**Comment by VladyslavBondarenko** [comment]
Seem to be solved, tried with `dev v1.16.4-pre.33`, closing

<details>
 <summary>code</summary>

```
import 'package:flutter/cupertino.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      home: HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return const CupertinoPageScaffold(
      child: CustomScrollView(
        slivers: <Widget>[
          CupertinoSliverNavigationBar(
            largeTitle: Text("Title"),
          ),
          SliverFillRemaining(
            child: Center(
              child: Text("Swipe down"),
            ),
          )
        ],
      ),
    );
  }
}
```

</details>

<details>
 <summary>screen recording</summary>
 
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/17024977/78253763-8c713a00-74fd-11ea-98ee-5e0c94e21dab.gif)

</details>

@jslavitz comment please If it's not so and I'll reopen the issue
